playground/web/index.html

```diff
@@ -0,0 +1,382 @@
+            background: #2c3e50;
+            color: #ecf0f1;
+            padding: 1rem;
+            border-radius: 4px;
+            white-space: pre-wrap;
+            font-family: 'Courier New', monospace;
+            font-size: 0.8rem;
+            line-height: 1.3;
+            overflow-y: auto;
+        }
+
+        .loading {
+            opacity: 0.6;
+            pointer-events: none;
+        }
+
+        @media (max-width: 768px) {
+            .container {
+                grid-template-columns: 1fr;
+            }
+            
+            .flags-container {
+                grid-template-columns: repeat(auto-fit, minmax(120px, 1fr));
+            }
+        }
+
+        footer {
+            text-align: center;
+            padding: 1rem;
+            background-color: #2c3e50;
+            color: #ecf0f1;
+            font-size: 0.9rem;
+            margin-top: 2rem;
+        }
+
+        footer a {
+            color: #ecf0f1;
+            text-decoration: none;
+        }
+
+        #highlighted-text {
+            white-space: pre-line;
+        }
+    </style>
+</head>
+<body>
+    <div class="header">
+        <h1>🎯 fancy-regex Playground</h1>
+        <p>Test and explore regular expressions with advanced features like backreferences and lookaround</p>
+    </div>
+
+    <div class="container">
+        <!-- Input Panel -->
+        <div class="panel">
+            <div class="panel-header">Input</div>
+            <div class="panel-content">
+                <div class="input-group">
+                    <label for="regex-input">Regular Expression:</label>
+                    <textarea id="regex-input" placeholder="Enter your regex pattern here, e.g., (?&lt;word&gt;\w+)\s+\k&lt;word&gt;"></textarea>
+                </div>
+
+                <div class="input-group">
+                    <label>Flags:</label>
+                    <div class="flags-container">
+                        <div class="flag-group">
+                            <input type="checkbox" id="flag-case-insensitive">
+                            <label for="flag-case-insensitive">Case Insensitive</label>
+                        </div>
+                        <div class="flag-group">
+                            <input type="checkbox" id="flag-multi-line">
+                            <label for="flag-multi-line">Multi-line</label>
+                        </div>
+                        <div class="flag-group">
+                            <input type="checkbox" id="flag-dot-matches-newline">
+                            <label for="flag-dot-matches-newline">Dot matches newline</label>
+                        </div>
+                        <div class="flag-group">
+                            <input type="checkbox" id="flag-ignore-whitespace">
+                            <label for="flag-ignore-whitespace">Ignore whitespace</label>
+                        </div>
+                        <div class="flag-group" style="display: none;">
```

Comment:
`display: none;` is one way to remove this 😆.. Can you clean this and the corresponding code up?